recipe-hooks-project/src/Recipes.js

```diff
@@ -0,0 +1,262 @@
+      headers: {
+        Authorization: "Token " + token
+      }
+    });
+    const ingredients = response.data;
+    setIngredients(ingredients);
+  }
+
+  async function fetchTags() {
+    console.log("Fetching tags for: [" + token + "]");
+    const response = await axios.request({
+      url: RECIPE_API_BASE_URL + TAGS_RELATIVE_URL,
+      method: "get",
+      headers: {Authorization: "Token " + token}
+    });
+    const tags = response.data;
+    setTags(tags);
+  }
+
+  async function handleCreateRecipe(e) {
+    e.preventDefault();
+    const response = await axios.request({
+      url: RECIPE_API_BASE_URL + RECIPES_RELATIVE_URL,
+      method: "post",
+      data: {
+        "title": title,
+        "ingredients": selectedIngredients,
+        "tags": selectedTags,
+        "time_minutes": time,
+        "price": price,
+        "link": link
+      },
+      headers: {Authorization: "Token " + token}
+    });
+    if (response.status === 201) {
+      toggleCreatingNewRecipe();
+      console.log("Success");
+    } else
+      console.log("Error");
+  }
+
+  function handleSelectIngredients(e) {
+    const data = e.target.options;
+    const selected = [];
+    for (let i = 0; i < data.length; i++) {
+      if (data[i].selected)
+        selected.push(data[i].value);
+    }
+    setSelectedIngredients(selected);
+  }
+
+  function handleSelectTags(e) {
+    const data = e.target.options;
+    const selected = [];
+    for (let i = 0; i < data.length; i++) {
+      if (data[i].selected)
+        selected.push(data[i].value);
+    }
+    setSelectedTags(selected);
+  }
+
+  return (
+    <Wrapper id="recipesWrapper">
+      <Title>
+        <div>Your favorite recipes</div>
+      </Title>
+
+      {
+        isCreatingNewRecipe
+          ? <>
+            <RecipeForm>
+              <RecipeFormSection>
+                <Label htmlFor="nameInput">Title: </Label>
+                <Input type="text" id="nameInput" value={title} onChange={handleTitleChange}/>
+              </RecipeFormSection>
+              <RecipeFormSection>
+                <Label htmlFor="selectIngredients">Ingredients</Label>
+                <RecipeFormSelect id="selectIngredients" onChange={handleSelectIngredients} multiple>
+                  {
+                    ingredientsList.map((item, i) => {
+                      return (
+                        <option key={i} value={item.id}>{item.name}</option>
+                      )
+                    }, this)}
+                </RecipeFormSelect>
+              </RecipeFormSection>
+              <RecipeFormSection>
+                <Label htmlFor="selectTags">Tags</Label>
+                <RecipeFormSelect id="selectTags" onChange={handleSelectTags} multiple>
+                  {
+                    tagsList.map((item, i) => {
+                      return (
+                        <option key={i} value={item.id}>{item.name}</option>
+                      )
+                    }, this)}
```

Comment:
You don't need the `this` argument here, because:
- you're in a functional component and not a class, so its value would be pretty useless
- you're using an arrow function which wouldn't rebind `this` https://www.freecodecamp.org/news/when-and-why-you-should-use-es6-arrow-functions-and-when-you-shouldnt-3d851d7f0b26/#main-benefit-no-binding-of-this-